spd/scripts/run.py

```diff
@@ -330,12 +336,45 @@ def generate_commands(
 
                 # Print first combination as example
                 if i == 0:
-                    print(f"  {experiment}: {len(combinations)} tasks")
-                    print(f"    Example params: {param_combo}")
+                    logger.info(f"  {experiment}: {len(combinations)} tasks")
+                    logger.info(f"    Example param overrides: {param_combo}")
+
+    if task_breakdown:
+        logger.values(task_breakdown)
 
     return commands
 
 
+def run_commands_locally(commands: list[str]) -> None:
+    """Execute commands locally in sequence.
+
+    Args:
+        commands: List of shell commands to execute
+    """
+    import shlex
```

Comment:
I think I prefer just importing this at the top of the file with the rest of the imports. Especially because it's just a small library.